src/main/java/com/cupid/jikting/common/jwt/filter/JwtAuthenticationProcessingFilter.java

```diff
@@ -0,0 +1,94 @@
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@RequiredArgsConstructor
+@Slf4j
+public class JwtAuthenticationProcessingFilter extends OncePerRequestFilter {
+
+    private static final String NO_CHECK_URL = "/members/login";
+
+    private final JwtService jwtService;
+    private final MemberRepository memberRepository;
+
+    private GrantedAuthoritiesMapper authoritiesMapper = new NullAuthoritiesMapper();
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request,
+                                    HttpServletResponse response,
+                                    FilterChain filterChain) throws ServletException, IOException {
+        String refreshToken = jwtService.extractRefreshToken(request).filter(jwtService::isTokenValid).orElse(null);
+
+        if (refreshToken != null) {
+            checkRefreshTokenAndReIssueAccessToken(response, refreshToken);
+            return;
+        }
+
+        if (refreshToken == null) {
+            checkAccessTokenAndAuthentication(request, response, filterChain);
+        }
+    }
+
+    public void checkRefreshTokenAndReIssueAccessToken(HttpServletResponse response, String refreshToken) {
+        memberRepository.findByRefreshToken(refreshToken).ifPresent(user -> {
+            String reIssuedRefreshToken = reIssueRefreshToken(user);
+            jwtService.sendAccessAndRefreshToken(response, jwtService.createAccessToken(user.getUsername()), reIssuedRefreshToken);
+        });
+    }
+
+    public void checkAccessTokenAndAuthentication(HttpServletRequest request,
+                                                  HttpServletResponse response,
+                                                  FilterChain filterChain) throws ServletException, IOException {
+        log.info("checkAccessTokenAndAuthentication() 호출");
+        jwtService.extractAccessToken(request).filter(jwtService::isTokenValid)
+                .flatMap(accessToken -> jwtService.extractUsername(accessToken)
+                        .flatMap(memberRepository::findByUsername))
+                .ifPresent(this::saveAuthentication);
+
+        filterChain.doFilter(request, response);
+    }
+
+    public void saveAuthentication(Member member) {
+        String password = member.getPassword();
+        UserDetails userDetailsUser = User.builder()
+                .username(member.getUsername())
+                .password(password)
+                .roles(member.getRole().name())
+                .build();
+        Authentication authentication = new UsernamePasswordAuthenticationToken(userDetailsUser,
+                null,
+                authoritiesMapper.mapAuthorities(userDetailsUser.getAuthorities()));
+
+        SecurityContextHolder.getContext().setAuthentication(authentication);
```

Comment:
p1. password는 한 번만 사용되는 변수여서 변수로 담지 않고 직접 사용하는게 더 좋을 것 같아요.

p2. UsernamePasswordAuthenticationToken에 `authenticated(Object principal, Object credentials, Collection<? extends GrantedAuthority> authorities)`라는 정적 팩토리 메소드가 있어서 이걸 사용하면 어떨까요?
동작은 생성자로 생성할 때와 똑같이 동작합니다!

p2. UsernamePasswordAuthenticationToken 생성 시 파라미터를 한 줄에 놓아도 눈에 잘 들어오는 것 같아요. 개행을 이런 방식으로 하는건 어떠신가요?

<img width="835" alt="스크린샷 2023-07-12 오후 3 43 14" src="https://github.com/SWM-Cupid/jikting-backend/assets/62989828/c5a2c7bc-307b-46d9-8014-62bb4838ea29">

```suggestion
        UserDetails userDetailsUser = User.builder()
                .username(member.getUsername())
                .password(member.getPassword())
                .roles(member.getRole().name())
                .build();
        Authentication authentication = UsernamePasswordAuthenticationToken.authenticated(
                userDetailsUser, null, authoritiesMapper.mapAuthorities(userDetailsUser.getAuthorities()));
        SecurityContextHolder.getContext().setAuthentication(authentication);
```